sampledb/logic/schemas/validate.py

```diff
@@ -310,36 +310,67 @@ def _validate_quantity(instance: dict, schema: dict, path: typing.List[str]) ->
     """
     if not isinstance(instance, dict):
         raise ValidationError('instance must be dict', path)
-    valid_keys = {'_type', 'units', 'dimensionality', 'magnitude_in_base_units'}
-    required_keys = valid_keys
+    valid_keys = {'_type', 'units', 'dimensionality', 'magnitude_in_base_units', 'magnitude'}
     schema_keys = set(instance.keys())
     invalid_keys = schema_keys - valid_keys
     if invalid_keys:
         raise ValidationError('unexpected keys in schema: {}'.format(invalid_keys), path)
-    missing_keys = required_keys - schema_keys
-    if missing_keys:
-        raise ValidationError('missing keys in schema: {}'.format(missing_keys), path)
+
+    #required_keys can no longer be statically defined, since either magnitude OR magnitude_in_base_units has to be given
```

Comment:
The code commented out below can probably be removed. Alternatively `required_keys` could be set to contain only `_type`, which is the only required key here.

Reply:
removed in update